rbx_types/src/qdir.rs

```diff
@@ -0,0 +1,43 @@
+use std::path::{Path, PathBuf};
+
+#[derive(Debug, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(transparent)
+)]
+pub struct QDir {
+    path_buf: PathBuf,
+}
+
+impl<'a> QDir {
+    pub fn new() -> Self {
+        QDir {
+            path_buf: PathBuf::new(),
+        }
+    }
+
+    pub fn path_buf(&'a self) -> &'a PathBuf {
+        &self.path_buf
+    }
```

Comment:
It feels more reasonable to return a `&Path` here rather than `&PathBuf`.

Reply:
This method is intended to be an accessor to the `path_buf` field (you can already get an `&Path` via the `AsRef<Path>` implementation). Thinking about it more, maybe this field doesn't really need an accessor?

Reply:
I'm not sure what the use is for a field accessor here. Maybe writing, though I don't believe this is how people generally do field mutability in Rust.